lsp-mode.el

```diff
@@ -1299,12 +1299,19 @@ On other systems, returns path without change."
   "Implemented only to make `company-lsp' happy.
 DELETE when `lsp-mode.el' is deleted.")
 
+(defconst url-path-allowed-chars-without-percent
+  (let ((vec (copy-sequence url-path-allowed-chars)))
+    (aset vec ?% nil)
```

Comment:
I think we can go further and instead of just removing `%`, remove all other additional delimiter characters except `/` from `url-path-allowed-chars`.
Since we are just hexifying local path, the union of `url-unresvered-chars` and `\` should be enough.
Something like https://github.com/kiennq/lsp-mode/commit/60947473381eb39fa0153fe3bb8680e27a2035a5
And it passes the travis build too.

Reply:
I'll try if this works with gopls. I tried with the default value before and it didn't work.

Reply:
Yes, I think it's because it normalizes the `/` so the path is not correct anymore.

Reply:
Yes, that was the problem iirc.